lib/rules/no-noop-setup-on-error-in-before.js

```diff
@@ -0,0 +1,137 @@
+'use strict';
+
+const { getImportIdentifier } = require('../utils/import');
+const types = require('../utils/types');
+
+//------------------------------------------------------------------------------
+// General rule - Disallows noop `setupOnError` in `before` or `beforeEach`.
+//------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'disallows using noop setupOnerror in `before` or `beforeEach`',
+      category: 'Testing',
+      recommended: false,
+      url:
+        'https://github.com/ember-cli/eslint-plugin-ember/tree/master/docs/rules/no-noop-setup-on-error-in-before.md',
+    },
+    fixable: 'code',
+    schema: [
+      {
+        type: 'object',
+        additionalProperties: false,
+      },
+    ],
+    messages: {
+      main: 'Using noop setupOnerror in `before` or `beforeEach` is not allowed',
+    },
+  },
+
+  create(context) {
+    let importedSetupOnerrorName, importedModuleName;
+    let isInModule = false;
+    let isInBeforeEachHook = false;
+    let isInBeforeHook = false;
+    let hooksIdentifier;
+    const sourceCode = context.getSourceCode();
+
+    /**
+     * Reports an error for a given node if we're currently in a `before` or `beforeEach` hook.
+     *
+     * @name reportErrorForNodeIfInBefore
+     * @param {Object} node
+     */
+    function reportErrorForNodeIfInBefore(node) {
+      const isInBeforeOrBeforeEach =
+        (isInBeforeEachHook || isInBeforeHook) && node.callee.name === importedSetupOnerrorName;
```

Comment:
Need to `types.isIdentifier(node.callee)` before checking `node.callee.name`. And there are several other places where type checks are missing that can lead to crashes or unexpected behavior.